modin/pandas/groupby.py

```diff
@@ -555,8 +593,14 @@ def size(self):
                 if "__reduced__" in result.columns
                 else result
             )
-        elif isinstance(self._df, Series):
-            result.name = self._df.name
+        elif self.ndim == 1:
+            result.name = (
+                self._df.name
```

Comment:
since this PR, `SeriesGroupbyObject` does not always contain a Series as a source object, that's why restoring a result name for the Series looks a bit tricky now

Reply:
This might be related to the comment on providing a subset of the columns.

Reply:
we can't get the Series object as a `self._df` anyway [[link]](https://github.com/modin-project/modin/pull/3298#discussion_r692283520), so in cases of 1-dim selection the "Series" name could be retrieved only from the selection, as I suppose